pandas/core/reshape/merge.py

```diff
@@ -362,8 +362,8 @@ def merge_asof(
 
     Parameters
     ----------
-    left : DataFrame
-    right : DataFrame
+    left : DataFrame or named Series
+    right : DataFrame or named Series
```

Comment:
DataFrame is mentioned quite often elsewhere in the docstring. Maybe should do a pass on the complete docstring for clarity.

Reply:
I started to change all references from "DataFrame" or "frame" to say "DataFrame or Series", and it became quite clumsy.  So I added a line at the top of the docs to say that a `Series` is treated as a `DataFrame` with a single named column, and then the rest of the docs works.